src/pydash/objects.py

```diff
@@ -1670,6 +1670,8 @@ def update_with(obj, path, updater, customizer=None):
         base_set(target, key, path_obj, allow_override=False)
 
         try:
+            if isinstance(key, str):
+                key = key.replace('.', r'\.')
             target = get(target, key)
```

Comment:
I think this could just be changed to `base_get(target, key, default=None)` instead of having to escape the dots.

Some test cases for this would be good too.